firebase-config/src/main/java/com/google/firebase/remoteconfig/RemoteConfigComponent.java

```diff
@@ -299,4 +324,43 @@ private static boolean isAbtSupported(FirebaseApp firebaseApp, String namespace)
   private static boolean isPrimaryApp(FirebaseApp firebaseApp) {
     return firebaseApp.getName().equals(FirebaseApp.DEFAULT_APP_NAME);
   }
+
+  private synchronized void notifyRCInstances(boolean background) {
+    for (FirebaseRemoteConfig frc : frcNamespaceInstancesBackground.values()) {
+      if (!background) {
+        // Add dummy listener and remove to trigger http stream flow.
+        ConfigUpdateListenerRegistration registration =
+            frc.addOnConfigUpdateListener(
+                new ConfigUpdateListener() {
+                  @Override
+                  public void onEvent() {}
+
+                  @Override
+                  public void onError(@NonNull Exception error) {}
+                });
+        registration.remove();
+      }
+    }
+  }
+
+  private class GlobalBackgroundListener
+      implements BackgroundDetector.BackgroundStateChangeListener {
+    private final AtomicReference<GlobalBackgroundListener> INSTANCE = new AtomicReference<>();
+
+    private void ensureBackgroundListenerIsRegistered(Context context) {
+      Application application = (Application) context.getApplicationContext();
+      if (INSTANCE.get() == null) {
+        GlobalBackgroundListener globalBackgroundListener = new GlobalBackgroundListener();
+        if (INSTANCE.compareAndSet(null, globalBackgroundListener)) {
+          BackgroundDetector.initialize(application);
+          BackgroundDetector.getInstance().addListener(globalBackgroundListener);
+        }
+      }
+    }
+
+    @Override
+    public void onBackgroundStateChanged(boolean b) {
```

Comment:
`b` is nondescriptive.

Reply:
updated to a more descriptive variable name. b to isInBackground